ffi/sfml-build/src/lib.rs

```diff
@@ -26,12 +26,16 @@ pub fn link_csfml(lib_name: &str) {
 
 // Add search path for SFML library files
 pub fn link_sfml(lib_name: &str) {
-    // SFML_HOME points to the base SFML directory
-    // Let cargo find the SFML library files there
-    if let Ok(sfml_home) = var("SFML_HOME") {
-        println!("cargo:rustc-link-search=native={}/lib", sfml_home);
-    }
+    // The windows build of CSFML links SFML libraries statically, so no
+    // need to link them to the rust executable as well.
```

Comment:
My primary concern is that I'm not sure if this is always the case. Is it the official download that does this? Does the GCC version do this too, or only the MSVC version? What about people who have built their SFML from source? Does it statically link by default when you build from source?

Perhaps we need to do more detailed analysis to determine these factors, and/or provide options.

Reply:
My belief is that how CSFML links to SFML is irrelevant, as long as CSFML itself is a DLL, which I understood was how this crate expects it to be. I'll try to compile CSFML in different ways and report back.

Reply:
As far as I can tell, my theory is correct: CSFML must be built as a dynamic library (I couldn't build it as static library), and how it links to SFML is irrelevant to this crate.

To build this crate you need the CSFML .lib files, and to run it you need the CSFML DLLs. If the CSFML DLLs were linked statically to SFML then all you need to run the programs are the CSFML DLLs; if CSFML links dynamically to SFML then you also need the SFML DLLs to run the program.

In either case, you never need the SFML .lib files to build or run this crate.

(This is how I would expect dynamic libs to work on any OS, but I don't really know specifics about the topic outside of Windows)

Reply:
Hmm. Actually, I tested it, and not linking against SFML works even on Linux.

Now I'm not sure anymore why we even linked against SFML in the first place.

Anyway, for now I'll just merge this, and maybe completely remove linking against SFML later, if it turns out there is no need for it.